theseus/labs/lie/lie_tensor.py

```diff
@@ -55,15 +56,30 @@ def euclidean_enabled() -> bool:
 
 
 class _LieTensorBase(torch.Tensor):
-    def __new__(cls, data: torch.Tensor, ltype: _ltype, requires_grad=False):
-        return cls._make_subclass(cls, data, requires_grad)  # type: ignore
+    def __new__(
+        cls,
+        data: torch.Tensor,
+        ltype: _ltype,
+        requires_grad: bool = False,
+        _shared_memory: bool = False,
+    ):
+        return cls._make_subclass(
+            cls, data if _shared_memory else data.clone(), requires_grad  # type: ignore
+        )
 
     @property
     def _t(self) -> torch.Tensor:
         return self.as_subclass(torch.Tensor)
 
-    def __init__(self, data: Any, ltype: _ltype, requires_grad=None):
+    def __init__(
+        self,
+        data: Any,
+        ltype: _ltype,
+        requires_grad: Optional[bool] = None,
+        _shared_memory: bool = False,
```

Comment:
So `_shared_memory` has not been used in the constructor? Also, I guess `_shared_memory` tries to use the initialized tensor?

Reply:
`_shared_memory` is used in `__new__`, which is called by Python before `__init__`, and it's used to allocate the memory for the new object.